lib/components/home/artist_rails/artist_rail.js

```diff
@@ -12,17 +12,42 @@ import ArtistCard from './artist_card'
 import Separator from '../../separator'
 
 class ArtistRail extends React.Component {
+  state: {
+    artists: [Object],
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      artists: []
+    }
+  }
 
   componentDidMount() {
     this.props.relay.setVariables({ fetchContent: true })
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.rail.results) {
+      this.setState({ artists: nextProps.rail.results.slice(0) })
```

Comment:
It’s a JS convention to create a copy of an array. Normally you would use `slice` to get a subarray starting from an element at a certain index, so using 0 means you get all of the elements in a new array.